zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -323,21 +418,51 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             offset,
             Value::known(F::from(tx.call_data_gas_cost)),
         )?;
+        self.nonce
+            .assign(region, offset, Value::known(F::from(tx.nonce + 1)))?;
+        self.nonce_prev
+            .assign(region, offset, Value::known(F::from(tx.nonce)))?;
```

Comment:
Oh didn't notice this before, but these values need to come from the account, not the transaction!
I'm also not sure how this works because the nonce stored in the account shouldn't always increment anymore for invalid transactions so the account nonce should remain the same.

Reply:
Seems like we need some additional changes to the witness generation: https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/b77613e9b42eacb03ccc894f0b353c374da3f283/bus-mapping/src/evm/opcodes.rs#L349-L358

Reply:
Does it mean we also need to change bus-mapping?